### PR TITLE
Implement Vertical Offset and slope transformation method

### PR DESCRIPTION
**Pull request by rouault**
@kbevers  I was not sure how to map the latitude and longitude of the "evaluation point" of the method, which is the 'origin' of the inclinated plane, to PROJ string parameters. I've used the classic lon_0 and lat_0 parameters, but not completely sure it is appropriate (I have to re-add lon_0 in the forward method, since generic code in PROJ, substracts it, as it is what is convenient for map projections)

### CODE REVIEW

File: docs/source/operations/transformations/vertoffset.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+| **Domain**          | 3D                                                       |
++---------------------+----------------------------------------------------------+
+| **Input type**      | Geodetic coordinates (horizontal), meters (vertical)     |
++---------------------+----------------------------------------------------------+
+| **output type**     | Geodetic coordinates (horizontal), meters (vertical)     |
++---------------------+----------------------------------------------------------+
+
+It is documented as coordinate operation method code 1046 in the EPSG dataset (:cite:`IOGP2018`).
+It is typically used in Europe, to relate national vertical systems to
+pan-European vertical systems (EVRF200, EVRF2007).
```
Comment (kbevers) [inline docs/source/operations/transformations/vertoffset.rst:25]: I would like to see a more in-depth description of the transformation here. It's too much of a faf to have to look up what 1046 means in the EPSG-world.

File: src/transformations/vertoffset.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    double nu0;
+};
+} // anonymous namespace
+
+// Cf EPSG Dataset coordinate operation method code 1046 "Vertical Offset and Slope"
+
+static PJ_XYZ forward_3d(PJ_LPZ lpz, PJ *P) {
+    const struct pj_opaque_vertoffset *Q = (const struct pj_opaque_vertoffset *) P->opaque;
+    PJ_XYZ xyz;
+    xyz.x = lpz.lam + P->lam0;
```
Comment (kbevers) [inline src/transformations/vertoffset.cpp:60]: Since this is a bit unusual a comment explaining what's going on would be nice

**Review by kbevers — COMMENTED** [review]
> I was not sure how to map the latitude and longitude of the "evaluation point" of the method, which is the 'origin' of the inclinated plane, to PROJ string parameters. I've used the classic lon_0 and lat_0 parameters, but not completely sure it is appropriate (I have to re-add lon_0 in the forward method, since generic code in PROJ, substracts it, as it is what is convenient for map projections)

I think the use of `+lat_0` and `+lon_0` fits the bill here. The meaning is of course a little bit different but I think it works alright in this case.

File: docs/source/operations/transformations/vertoffset.rst (hunk 1 above)
Comment (rouault) [reply docs/source/operations/transformations/vertoffset.rst:25]: there's isn't much text to borrow from the description of the EPSG dataset. I've added a Formula section which is the "objective" description of what this method does

File: src/transformations/vertoffset.cpp (hunk 2 above)
Comment (rouault) [reply src/transformations/vertoffset.cpp:60]: comment added

File: docs/source/operations/transformations/vertoffset.rst (hunk 1 above)
Comment (kbevers) [reply docs/source/operations/transformations/vertoffset.rst:25]: Thanks, I think that puts the docs in a good place

**Review by kbevers — APPROVED** [review]

**Comment by rhuijben** [comment]
Can you update the in c declaration header to include the required arguments of the projection, like we have in the other projection definitions?

In my Proj use I try to automatically create some test scenario's based on those. And it would also be nice for the built-in help from the Proj tools.

**Comment by rouault** [comment]
> Can you update the in c declaration header to include the required arguments

done, although as I mentioned in a commit comment, we haven't done that for transformation operations until now, just map projections

**Pull request merged**